modules/component/src/let/let.directive.ts

```diff
@@ -168,11 +159,14 @@ export class LetDirective<U> implements OnDestroy {
     this.subscription = this.cdAware.subscribe({});
   }
 
-  createEmbeddedView() {
-    this.embeddedView = this.viewContainerRef.createEmbeddedView(
-      this.templateRef,
-      this.viewContext
-    );
+  ensureEmbeddedViewCreated() {
+    if (!this.embeddedviewCreated) {
+      this.embeddedviewCreated = true;
+      this.viewContainerRef.createEmbeddedView(
+        this.templateRef,
+        this.viewContext
+      );
+    }
```

Comment:
Removing public `createEmbeddedView` method is a breaking change (if someone uses `LetDirective` outside of the template 👀). Let's keep it for now and remove `ensureEmbeddedViewCreated`:

```ts
  createEmbeddedView() {
    this.isEmbeddedViewCreated = true;
    this.viewContainerRef.createEmbeddedView(
      this.templateRef,
      this.viewContext
    );
  }
```

Reply:
Good catch, I put the logic back to the way it was before except for changing order of setting the view context before rendering. Otherwise, the test I added would fail.

On a side point, outside the scope of this PR.
Would it make sense to deprecate and remove the function from the public API in the future?
If you call the function from your component the end result would be the content will get rendered twice. It's very hard to believe that would be desirable behavior.
The one use case that I can see where calling it yourself may be helpful is if you want your template to render even though the observable hadn't yet emitted. To support that scenario we can still stop you from rendering a second time by moving the `if(isEmbeddedviewCreated)`  logic to `createEmbeddedView`. Alternatively we can offer a config option to support this scenario.
We can create a new feature request if that makes sense.

Reply:
Yes, I agree that the `createEmbeddedView` method should not be public. Extending the `LetDirective` is not recommended approach at all. However, someone can do that, and we have to ensure that there are no breaking changes in minor/patch version updates.